modules/coinswap/keeper/keeper.go

```diff
@@ -103,13 +104,22 @@ func (k Keeper) Swap(ctx sdk.Context, msg *types.MsgSwapOrder) error {
 }
 
 // AddLiquidity adds liquidity to the specified pool
-func (k Keeper) AddLiquidity(ctx sdk.Context, msg *types.MsgAddLiquidity) (sdk.Coin, error) {
+func (k Keeper) AddLiquidity(ctx sdk.Context, imsg interface{}) (sdk.Coin, error) {
+	var msg *types.MsgAddLiquidity
+
+	switch i := imsg.(type) {
+	case *types.MsgAddUnilateralLiquidity:
+		return k.addUnilateralLiquidity(ctx, i)
+	case *types.MsgAddLiquidity:
+		msg = i
+	}
+
 	standardDenom := k.GetStandardDenom(ctx)
```

Comment:
why not add function `AddUnilateralLiquidity`

Reply:
Proposed to do that way before but was asked to just add an interface assertion. Maybe refactor this now?